src/Microsoft.DotNet.Wpf/src/PresentationUI/MS/Internal/documents/DocumentSignatureManager.cs

```diff
@@ -506,9 +506,7 @@ internal void ShowSigningDialog(IntPtr parentWindow)
                                             // Get[Algorithm]PrivateKey methods would always have returned the private key if the PrivateKey property would
                                             // But Get[Algorithm]PrivateKey methods never throw but returns null in case of error during cryptographic operations
                                             // But we want exception to be thrown when an error occurs during a cryptographic operation so that we can redisplay the certificate picker
-                                            #pragma warning disable SYSLIB0028
-                                            AsymmetricAlgorithm testKey = x509Certificate2.PrivateKey;
-                                            #pragma warning restore SYSLIB0028
+                                            throw new CryptographicException("PrivateKey is not retrievable.");
```

Comment:
Same concerns as my remark in XmlDigitalSignatureProcessor.cs.  Except this exception is caught 8 lines down, so it's not visible to user.

Reply:
The try/catch could probably be removed and just set requestAgain to true instead of throwing since none of the Get*Key methods are documented to throw CryptographicException.